addon/serializers/web-api.js

```diff
@@ -45,22 +48,32 @@ export default DS.RESTSerializer.extend({
     }
   },
 
-  extractErrors: function(store, typeClass, payload, id) {
-    let modelState = payload.modelState,
-        payloadKey = `${typeClass.modelName}.`,
-        formattedPayload = { errors: { } },
-        key;
-
-    for(key in modelState) {
-      if(modelState.hasOwnProperty(key)) {
-        formattedPayload.errors[key.replace(payloadKey, '').camelize()] = modelState[key];
-      }
-    }
-
-    return this._super(store, typeClass, formattedPayload, id);
+  normalizeErrors: function normalizeErrors(typeClass, payload) {
+    let payloadKey = `${typeClass.modelName}.`,
+      key,
```

Comment:
I'm nit-picking here, but the indentation needs to be fixed here (should be aligned with `payloadKey`.